src/graph.rs

```diff
@@ -138,6 +163,13 @@ impl RenderThread {
                     *sample = value;
                 }
             }
+
+            if data.len() != chunk_size {
+                // this is the last chunk, and it contained less than BUFFER_SIZE samples
+                let channel_offset = data.len() / self.channels;
+                debug_assert!(channel_offset < BUFFER_SIZE as usize);
+                self.buffer_offset = Some((channel_offset, rendered.clone()));
```

Comment:
Due to this clone(), I don't think the current implementation is copy free (afaik).

Possible solutions:
- Make clone operation cheap by wrapping AudioBuffer into a ref counter
- Add a specific api in Audiobuffer to call when not consumed buffer is detected (e.g channel_data_with_offset). It would required to move the "not consumed" AudioBuffer into RenderThread.buffer_offset. this move maybe also costly (need to be measured)

Disclaimer: i didn't try what i propose and i could be completely wrong here.

Reply:
The `alloc::AudioBuffer` is just a bunch of `Rc`s, so it is cheap to clone (only increment the ref count)